distributionviewer/core/static/js/app/components/containers/chart-container.js

```diff
@@ -58,20 +50,14 @@ class ChartContainer extends React.Component {
       this.refLabels[item.x] = item.label;
     });
 
-    // Category charts get treated differently since they start at x: 1
-    if (props.metric.type === 'category') {
-      this.xScale = d3Scale.scaleLinear()
-                      .domain([1, d3Array.max(this.data, d => d.x)])
-                      .range([0, this.size.innerWidth]);
-    } else {
-      this.xScale = d3Scale.scaleLinear()
-                      .domain([0, d3Array.max(this.data, d => d.x)])
-                      .range([0, this.size.innerWidth]);
-    }
-
     this.yScale = d3Scale.scaleLinear()
                     .domain([0, d3Array.max(this.data, d => d.y)])
-                    .range([this.size.innerHeight, 0]);
+                    .range([this.state.size.innerHeight, 0]);
+
+    this._setWidth(props);
+    if (props.isDetail) {
+      window.addEventListener('resize', this._handleResize);
```

Comment:
We should debounce this with something like https://davidwalsh.name/javascript-debounce-function because it will potentially fire too often. We used a 250ms timer on marketplace that worked well.